### PR TITLE
Add support for phpredis cache adapter

### PR DESCRIPTION
**Pull request by yellow1912**
This is basically a clone of Predis adapter code with some tweaks to make it work with phpredis

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
You need to add this to the suggested dependencies in the composer.json.

**Comment by GrahamCampbell** [comment]
> This is basically a clone of Predis adapter code

You must attribute the code correctly as per its license.

File: src/Storage/PhpRedis.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    /**
+     * @var int|null seconds until cache expiration
+     */
+    protected $expire;
+
+    /**
+     * Constructor.
+     *
+     * @param \Redis $client phpredis client
```
Comment (GrahamCampbell) [inline src/Storage/PhpRedis.php]: phpdoc is wrong - missing the `|null` bit

File: src/Storage/PhpRedis.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+     * @param string         $key    storage key
```
Comment (GrahamCampbell) [inline src/Storage/PhpRedis.php]: phpdoc is wrong - missing the `|null` bit

**Comment by yellow1912** [comment]
I added the suggestion for phpredis, please note that it is a php extension.

I add the null in phpdoc (note that the original Predis class also missing it)

Not sure how should I "attribute the code correctly as per its license", I copied the code directly from the Predis class in this bundle.

**Comment by GrahamCampbell** [comment]
Oh, I thought you meant you'd copied code out of "predis".

**Comment by GrahamCampbell** [comment]
https://github.com/nrk/predis

**Comment by frankdejonge** [comment]
@GrahamCampbell I'm pretty sure he meant it's copied from the PredisAdapter which resides in _this_ package.

**Comment by GrahamCampbell** [comment]
I see that, yeh. :)

**Comment by frankdejonge** [comment]
@yellow1912 thanks you for your contribution, I've tagged the 1.0.2 release so this can ship now.

**Pull request merged**